AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/edit-work-hours.component.ts

```diff
@@ -99,13 +101,24 @@ export class EditWorkHoursComponent implements OnInit {
         this.result = $event;
         this.showWorkHoursTable = false;
         this.showNonWorkHoursTable = false;
-        if (this.result) {
-            this.showWorkHoursTable = this.result[0] instanceof VhoWorkHoursResponse;
-            this.showNonWorkHoursTable = this.result[0] instanceof VhoNonAvailabilityWorkHoursResponse;
+        switch (this.hoursType) {
+            case HoursType.WorkingHours:
+                this.showWorkHoursTable = true;
+                this.showNonWorkHoursTable = false;
+                break;
+            case HoursType.NonWorkingHours:
+                this.showNonWorkHoursTable = true;
+                this.showWorkHoursTable = false;
```

Comment:
as per previous comment

Reply:
just to be on the safe side if it will be called after an external event